enhancements/monitoring/alerts-ui-management.md

```diff
@@ -0,0 +1,336 @@
+
+4. **Create alert from a metric query**
+   As a developer, I want to write or paste a PromQL expression in a “Create Alert” form, specify duration and severity, and save it,
+   so that I can turn any ad‑hoc query I used during troubleshooting into a managed alert.
+
+5. **View silence status on active alerts**
+   As a cluster admin, I want the “Active Alerts” list to show which alerts are currently silenced (and until when),
+   so that I have clear visibility into which notifications are suppressed and why.
+
+## Goals
+1. CRUD operations on user‑defined alerts via UI and API.
+2. Clone platform or user alerts to customize thresholds or scopes.
+3. Disable/enable alerts by creating/updating entries in the `AlertRelabelConfig` CR.
+4. Create/manage silences in Alertmanager and reflect this in the UI.
+5. Aggregate view of all alerting rules, showing definitions plus relabel context.
+6. Aggregate view of all alerts, showing status (Pending, Firing, Silenced) and relabel context.
+7. Enforce multi‑tenancy: restrict queries to user’s namespace or cluster scope.
+8. Persist user changes through operator upgrades, without modifying existing operators or CRDs.
+9. Stay fully GitOps/Argo CD compliant: managed resources must remain commit‑able and, when owned by a GitOps application, appear read‑only.
+
+## Non‑Goals
+- Deep RBAC beyond native Kubernetes permissions.
+- Operators reacting to user modifications (operator code remains unchanged).
+- Full multi‑cluster federation (initial focus is single‑cluster).
+
+## Proposal
+
+### 1. API Endpoints
+
+| Method  | Path                                             | Description                                                                          |
+|---------|--------------------------------------------------|--------------------------------------------------------------------------------------|
+| GET     | `/alerting/rules`                                | List all alert rule definitions, including `disabled` flags and relabel context.     |
+| GET     | `/alerting/alerts`                               | List active alerts (Pending / Firing / Silenced), showing post‑relabel data.         |
+| POST    | `/alerting/rules`                                | Create a new user‑defined alert rule.                                                |
+| PATCH   | `/alerting/rules/{alertname}/{severity}`         | Partially update a single rule (e.g. toggle `disabled`, tweak expression).           |
+| PATCH   | `/alerting/rules`                                | Bulk partial updates of multiple rules in one atomic request (each with its own `alertname`, `severity`, and fields to change). |
+| DELETE  | `/alerting/rules/{alertname}/{severity}`         | Delete a user‑defined rule identified by its `alertname` and post‑relabel `severity`.|
+| POST    | `/alerting/rules/{alertname}/{severity}/clone`   | Clone an existing rule for quick customization (new name, adjusted severity, etc.).  |
```

Comment:
`{alertname}/{severity}` is not common and the url usage will make no sense

all routes should support the severity as a param, and not a URL field in my opinion

Reply:
```suggestion
| GET       | `/api/alerting/alerts` | List active alerts (Pending / Firing / Silenced) |

| GET       | `/api/alerting/rules` | List all alerting rules definitions |
| POST     | `/api/alerting/rules` | Create a new user‑defined alerting rule. |
| PUT       | `/api/alerting/rules/{alertname}` | Update an existing user‑defined alerting rule |
| DELETE | `/api/alerting/rules/{alertname}` | Delete a user‑defined alerting rule |

| POST     | `/api/alerting/rules/{alertname}/disable` | Disable an alerting rule |
| POST     | `/api/alerting/rules/{alertname}/enable` | Enable an alerting rule |

| POST     | `/api/alerting/rules/{alertname}/clone` | Clone an existing alerting rule |

| GET        | `/api/health` | Health check endpoint |
```

Reply:
<img width="746" height="849" alt="Screenshot from 2025-07-28 14-17-11" src="https://github.com/user-attachments/assets/ddd97291-c9ef-430f-baa8-71c5a6c2d01b" />